packages/storybook/src/plugins/plugin.ts

```diff
@@ -0,0 +1,370 @@
+import {
+  CreateDependencies,
+  CreateNodes,
+  CreateNodesContext,
+  TargetConfiguration,
+  detectPackageManager,
+  joinPathFragments,
+  parseJson,
+  readJsonFile,
+  workspaceRoot,
+  writeJsonFile,
+} from '@nx/devkit';
+import { dirname, isAbsolute, join, relative } from 'path';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+import { existsSync, readFileSync, readdirSync } from 'fs';
+import { calculateHashForCreateNodes } from '@nx/devkit/src/utils/calculate-hash-for-create-nodes';
+import { projectGraphCacheDirectory } from 'nx/src/utils/cache-directory';
+import { getLockFileName } from '@nx/js';
+import { tsquery } from '@phenomnomnominal/tsquery';
```

Comment:
I wonder if `tsquery` is the way to go, or can we import the file like we do in other plugins. Is there a reason querying is preferred?

Reply:
requiring it gives a 
```
import type { StorybookConfig } from '@storybook/angular';
   ^^^^^^
SyntaxError: Cannot use import statement outside a module
```
error.

AST is not that much of a performance concern in this case:
```
Time for '@nx/storybook/plugin:createNodes' 162.18487500026822
Time for '@nx/storybook/plugin:createNodes:apps/ngapp1/.storybook/main.ts' 0.4941249992698431
Time for '@nx/storybook/plugin:createNodes:apps/rapp1/.storybook/main.ts' 0.4677499998360872
Time for '@nx/storybook/plugin:createNodes:ngapp2/.storybook/main.ts' 0.37250000052154064
Time for '@nx/storybook/plugin:createNodes:rapp3/.storybook/main.ts' 0.39054099842906
Time for '@nx/storybook/plugin:createNodes:rapp4/.storybook/main.ts' 0.33058400079607964

my logs just for getStorybookConfig
getStorybookConfig took 5.05587499961257 milliseconds
getStorybookConfig took 1.6850829999893904 milliseconds
getStorybookConfig took 0.6668329983949661 milliseconds
getStorybookConfig took 0.5654579997062683 milliseconds
getStorybookConfig took 0.5120829995721579 milliseconds

for cypress, which requires the module
Time for '@nx/cypress/plugin:createNodes' 158.9198329988867
Time for '@nx/cypress/plugin:createNodes:rapp3-e2e/cypress.config.ts' 0.40424999967217445
Time for '@nx/cypress/plugin:createNodes:rapp4-e2e/cypress.config.ts' 0.30654199980199337
```